# HW 2



### Step 1: Load the python libraries you will need for this project 

In [8]:
#imports
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

%matplotlib inline

### Step 2: Read in your data set 

In [16]:
#Read in data from source 
url = "https://raw.githubusercontent.com/ga-students/DS-SF-24/master/Data/admissions.csv"
df_raw = pd.read_csv(url)
df_raw.head(100)

admit    gre   gpa  prestige
0       0  380.0  3.61       3.0
1       1  660.0  3.67       3.0
2       1  800.0  4.00       1.0
3       1  640.0  3.19       4.0
4       0  520.0  2.93       4.0
5       1  760.0  3.00       2.0
6       1  560.0  2.98       1.0
7       0  400.0  3.08       2.0
8       1  540.0  3.39       3.0
9       0  700.0  3.92       2.0
10      0  800.0  4.00       4.0
11      0  440.0  3.22       1.0
12      1  760.0  4.00       1.0
13      0  700.0  3.08       2.0
14      1  700.0  4.00       1.0
15      0  480.0  3.44       3.0
16      0  780.0  3.87       4.0
17      0  360.0  2.56       3.0
18      0  800.0  3.75       2.0
19      1  540.0  3.81       1.0
20      0  500.0  3.17       3.0
21      1  660.0  3.63       2.0
22      0  600.0  2.82       4.0
23      0  680.0  3.19       4.0
24      1  760.0  3.35       2.0
25      1  800.0  3.66       1.0
26      1  620.0  3.61       1.0
27      1  520.0  3.74       4.0
28      1  780.0  3.22       2.0
29      0  520.0  3.29       1.0
..    ...    ...   ...       ...
70      0  640.0  4.00       3.0
71      0  300.0  2.92       4.0
72      0  480.0  3.39       4.0
73      0  580.0  4.00       2.0
74      0  720.0  3.45       4.0
75      0  720.0  4.00       3.0
76      0  560.0  3.36       3.0
77      1  800.0  4.00       3.0
78      0  540.0  3.12       1.0
79      1  620.0  4.00       1.0
80      0  700.0  2.90       4.0
81      0  620.0  3.07       2.0
82      0  500.0  2.71       2.0
83      0  380.0  2.91       4.0
84      1  500.0  3.60       3.0
85      0  520.0  2.98       2.0
86      0  600.0  3.32       2.0
87      0  600.0  3.48       2.0
88      0  700.0  3.28       1.0
89      1  660.0  4.00       2.0
90      0  700.0  3.83       2.0
91      1  720.0  3.64       1.0
92      0  800.0  3.90       2.0
93      0  580.0  2.93       2.0
94      1  660.0  3.44       2.0
95      0  660.0  3.33       2.0
96      0  640.0  3.52       4.0
97      0  480.0  3.57       2.0
98      0  700.0  2.88       2.0
99      0  400.0  3.31       3.0

[100 rows x 4 columns]

## Questions
#### Question 1. How many observations are in our dataset? 


In [10]:
len(df_raw)

400

Answer: 400

#### Question 2. Create a summary table

In [11]:
df_raw.describe()

admit         gre        gpa    prestige
count  400.000000  398.000000  398.00000  399.000000
mean     0.317500  588.040201    3.39093    2.486216
std      0.466087  115.628513    0.38063    0.945333
min      0.000000  220.000000    2.26000    1.000000
25%      0.000000  520.000000    3.13000    2.000000
50%      0.000000  580.000000    3.39500    2.000000
75%      1.000000  660.000000    3.67000    3.000000
max      1.000000  800.000000    4.00000    4.000000

#### Question 3. Why would GRE have a larger STD than GPA?

Answer: they are on different scales. gre_min = 220 gre_max = 800 and gpa_min = 2.26 gpa_max = 4.00

#### Question 4. Drop data points with missing data


In [12]:
df_raw.dropna(inplace = True)

#### Question 5. Confirm that you dropped the correct data. How can you tell? 

In [14]:
len(df_raw.isnull())
# 400 - 397 = 3 

397

#### Question 6. Create box plots for GRE and GPA

In [ ]:
#boxplot 1 GRE


In [ ]:
#boxplot 2 GPA



#### Question 7. What do these plots show?

Answer: 

#### Question 8.  If our model had an assumption of a symmetric distribution would we meet that requirement? 

Answer: 

#### Question 9: Add dummy variables for GRE scores. Top 10% are considered very high. next 25% of GRE scores are considered high, next 50% are considered medium and lowest 15% are considered Low. 

#### By a scatterplot explore the association between GRE based on GPA. For each observation, color according to perstige. 